### PR TITLE
[ApplicationRequestSpace] refacto municipal spaces

### PR DESCRIPTION
**Pull request by ottaviano**

### CODE REVIEW

File: src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            ]
+        ));
+    }
+
+    protected function redirectToSpaceRoute(string $subName, array $parameters = []): Response
+    {
+        return $this->redirectToRoute("app_application_request_{$this->getSpaceName()}_${subName}", $parameters);
+    }
+
+    protected function checkAccess(?ApplicationRequest $request): void
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php]: Maybe we can abstract this method, WDYT ?

File: src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+use AppBundle\Controller\CanaryControllerTrait;
+use AppBundle\Entity\ApplicationRequest\ApplicationRequest;
+use AppBundle\Form\ApplicationRequest\ApplicationRequestTagsType;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+abstract class AbstractController extends Controller
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php]: This name can be confuse with the `AbstractController` of the `FrameworkBundle`.

WDYT to rename it ?

File: src/Controller/EnMarche/ApplicationRequestCandidate/MunicipalChiefSpaceController.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+class MunicipalChiefSpaceController extends AbstractController
+{
+    protected function getApplicationRequests(ApplicationRequestRepository $repository, string $type): array
+    {
+        return $repository->findAllForInseeCodes($this->getUser()->getMunicipalChiefManagedArea()->getCodes(), $type);
+    }
+
+    protected function getSpaceName(): string
+    {
+        return 'municipal_chief';
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/ApplicationRequestCandidate/MunicipalChiefSpaceController.php]: Can you use `Enum` for the space name ?

File: src/Controller/EnMarche/ApplicationRequestCandidate/ReferentSpaceController.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+class ReferentSpaceController extends AbstractController
+{
+    protected function getApplicationRequests(ApplicationRequestRepository $repository, string $type): array
+    {
+        return $repository->findAllForReferentTags($this->getUser()->getManagedArea()->getTags()->toArray(), $type);
+    }
+
+    protected function getSpaceName(): string
+    {
+        return 'referent';
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/ApplicationRequestCandidate/ReferentSpaceController.php]: Same here for `Enum`.

File: src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php (hunk 1 above)
Comment (ottaviano) [reply src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php]: done

File: src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php (hunk 2 above)
Comment (ottaviano) [reply src/Controller/EnMarche/ApplicationRequestCandidate/AbstractController.php]: why not, done

File: src/Controller/EnMarche/ApplicationRequestCandidate/MunicipalChiefSpaceController.php (hunk 3 above)
Comment (ottaviano) [reply src/Controller/EnMarche/ApplicationRequestCandidate/MunicipalChiefSpaceController.php]: I added controller constant for this value, is it ok?

Comment (thomasbisignani) [reply src/Controller/EnMarche/ApplicationRequestCandidate/MunicipalChiefSpaceController.php]: :+1:

**Pull request merged**